ruma-appservice-api/src/event/push_events/v1.rs

```diff
@@ -41,6 +41,84 @@ impl IncomingRequest {
     pub fn new(txn_id: String, events: Vec<Raw<AnyEvent>>) -> IncomingRequest {
         IncomingRequest { txn_id, events }
     }
+
+    /// Consumes the `IncomingRequest` and tries to convert it to a `sync_events::Response`
+    ///
+    /// This is a helper conversion in cases where it's easier to work with `sync_events::Response`
+    /// instead of the original `push_events::IncomingRequest`. It puts all events with a `room_id`
+    /// into the `JoinedRoom`'s `timeline` or into `ephemeral` if no `state_key` and `event_id` is
+    /// set. The rationale behind that is that incoming Appservice transactions from the homeserver
+    /// are not necessarily bound to a specific user but can cover a multitude of namespaces, and
+    /// as such the Appservice basically only "observes joined rooms".
+    ///
+    /// Additionally events with type `m.presence` are put into the response's `presence`
+    /// accordingly.
+    #[cfg(feature = "helper")]
+    pub fn try_into_sync_response(
+        self,
+        next_batch: impl Into<String>,
+    ) -> Result<ruma_client_api::r0::sync::sync_events::Response, serde_json::Error> {
+        use ruma_client_api::r0::sync::sync_events;
+        use ruma_events::{presence::PresenceEvent, AnySyncEphemeralRoomEvent, AnySyncRoomEvent};
+        use ruma_identifiers::RoomId;
+        use serde::{de::IgnoredAny, Deserialize};
```

Comment:
Shouldn't these be moved to the top of the file?

Reply:
I've interpreted https://github.com/ruma/ruma/pull/514#discussion_r616486428 as such that the idea was to encapsulate everything for that feature in the inherent method

Reply:
It makes sense the way it is now. Exactly as to not require four additional cfg attributes for the imports and one for `EventDeHelper`.